ocaml/xapi/xapi_observer_components.ml

```diff
@@ -29,13 +29,13 @@ let to_string = function
       Constants.observer_component_smapi
 
 let of_string = function
-  | str when String.equal str Constants.observer_component_xapi ->
+  | str when str = Constants.observer_component_xapi ->
```

Comment:
`=` has compiler magic, I'd prefer if the specialized equal function was kept

Reply:
Reverted here https://github.com/xapi-project/xen-api/pull/5445/commits/55f9577f6a031a390f848af4dcfdb433932e78bf.